### PR TITLE
Add an example with Link

### PR DESCRIPTION
**Pull request by charliegerard-stripe**
This PR adds an example to showcase Link in the `custom-payment-flow` sample.

I updated the servers in all 7 programming languages but could do with some help with the PHP one. When the payment using Link is successful, it should redirect to `/success` and I couldn't figure out how to implement that.

Steps to review this PR:
1. Fetch this branch
2. Run one of the servers
3. Select the Link example
- [ ] The email field should be displayed
4. If you **don't** already have an account with Link, enter your email address.
   - [ ] A field should appear at the bottom of the form to save your information to pay faster with Link
   - Enter test credit card details and click `Pay`
   - Restart the process and enter your email
   - [ ] The field to submit an authentication code should appear
   - Enter `424242`
   - [ ] Your email should be recognized and the card details should be automatically filled
   - [ ] Click `Pay` and the payment should be successful

If you **do** have an account with Link, enter your email address. 
   - [ ] The field to submit an authentication code should appear
   - Enter `424242`
   - [ ] Your email should be recognized and the card details should be automatically filled
   - [ ] Click `Pay` and the payment should be successful

5. Repeat with each of the other servers
---

*Note: I only updated the `html` client for now to avoid making this PR too large.*

### CODE REVIEW

File: custom-payment-flow/client/html/link.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  const {publishableKey} = await fetch('/config').then((r) => r.json());
+  if (!publishableKey) {
+    addMessage(
+      'No publishable key returned from the server. Please check `.env` and try again'
+    );
+    alert('Please set your Stripe publishable API key in the .env file');
+  }
+
+  const stripe = Stripe(publishableKey, {
+    apiVersion: '2020-08-27',
```
Comment (cjavilla-stripe) [inline custom-payment-flow/client/html/link.js:13]: wonder if we should update all of these to the latest `2022-11-15` at some point.

File: custom-payment-flow/client/html/link.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  // Enable the skeleton loader UI for the optimal loading experience.
+  const loader = 'auto';
+
+  // Create an elements group from the Stripe instance passing in the clientSecret and enabling the loader UI.
+  const elements = stripe.elements({clientSecret, loader});
+
+  const linkAuthenticationElement = elements.create('linkAuthentication');
+
+  // Passing in defaultValues is optional, but useful if you want to prefill customer information to simplify the customer experience.
+  const paymentElement = elements.create('payment', {
```
Comment (cjavilla-stripe) [inline custom-payment-flow/client/html/link.js:38]: Nice 😎

File: custom-payment-flow/server/dotnet/Program.cs [hunk 3]
```diff
@@ -42,11 +42,12 @@
 
 app.MapPost("/create-payment-intent", async (CreatePaymentIntentRequest req) =>
 {
+    var formattedPaymentMethodType = req.PaymentMethodType == "link" ? new List<string> { "link", "card" } : new List<string> { req.PaymentMethodType };
```
Comment (cjavilla-stripe) [inline custom-payment-flow/server/dotnet/Program.cs:45]: ah bummer that we need to be able to pass two for link. hmm.

File: custom-payment-flow/server/java/src/main/java/com/stripe/sample/Server.java [hunk 4]
```diff
@@ -102,6 +102,10 @@ public static void main(String[] args) {
               .setCurrency(postBody.getCurrency())
               .setAmount(5999L);
 
+             if(postBody.getPaymentMethodType().equals("link")){
+                paramsBuilder.addPaymentMethodType("card");
```
Comment (cjavilla-stripe) [inline custom-payment-flow/server/java/src/main/java/com/stripe/sample/Server.java:106]: do we need to also call `paramsBuilder.addPaymentMethodType("link");` here?

File: custom-payment-flow/server/node-typescript/src/server.ts [hunk 5]
```diff
@@ -117,6 +117,23 @@ app.post(
   }
 );
 
+app.get('/payment/next', async (req, res) => {
+  const paymentIntent : any = req.query.payment_intent; 
+  const intent  = await stripe.paymentIntents.retrieve(
+    paymentIntent,
+    {
+      expand: ['payment_method'],
```
Comment (cjavilla-stripe) [inline custom-payment-flow/server/node-typescript/src/server.ts:125]: didn't see expansion in the other languages, do we want to also fetch the related payment method for the other server languages?

File: custom-payment-flow/server/node/server.js [hunk 6]
```diff
@@ -116,6 +116,22 @@ app.post('/create-payment-intent', async (req, res) => {
   }
 });
 
+app.get('/payment/next', async (req, res) => {
+  const intent = await stripe.paymentIntents.retrieve(
+    req.query.payment_intent,
+    {
+      expand: ['payment_method'],
```
Comment (cjavilla-stripe) [inline custom-payment-flow/server/node/server.js:123]: ditto on expand

File: custom-payment-flow/server/php/public/link.php [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Link</title>
+    <link rel="stylesheet" href="css/base.css" />
+    <script src="https://js.stripe.com/v3/"></script>
+    <script src="./utils.js"></script>
+    <script>
+      document.addEventListener('DOMContentLoaded', async () => {
+        const stripe = Stripe('<?= $_ENV["STRIPE_PUBLISHABLE_KEY"]; ?>', {
+          apiVersion: '2020-08-27',
```
Comment (cjavilla-stripe) [inline custom-payment-flow/server/php/public/link.php:38]: ditto on api version question, should we use latest?

**Review by cjavilla-stripe — APPROVED** [review]
Nice work! left a couple comments.

**Comment by mpbagwel** [comment]
@cjavilla-stripe @charliegerard-stripe Just following up on this one. Anything left to do before merging?

File: custom-payment-flow/client/html/link.js (hunk 1 above)
Comment (charliegerard-stripe) [reply custom-payment-flow/client/html/link.js:13]: I'm updating the link file for now and can open another issue to do it in other files

**Comment by charliegerard-stripe** [comment]
I implemented the feedback and tested again the different servers, should be good to merge.

**Pull request merged**